src/main/java/org/folio/edge/sip2/repositories/CirculationRepository.java

```diff
@@ -109,71 +115,143 @@ public Future<CheckinResponse> performCheckinCommand(Checkin checkin, SessionDat
+            .ok(FALSE)
+            .renewalOk(FALSE)
+            .magneticMedia(null)
+            .desensitize(FALSE)
+            .transactionDate(OffsetDateTime.now(clock))
+            .institutionId(institutionId)
+            .patronIdentifier(patronIdentifier)
+            .itemIdentifier(itemIdentifier)
+            .titleIdentifier(UNKNOWN)
+            .dueDate(OffsetDateTime.now(clock))
+            .screenMessage(verification.getErrorMessages())
+            .build());
+        }
+
+        final User user = verification.getUser();
+        final JsonObject body = new JsonObject()
+            .put("itemBarcode", itemIdentifier)
+            .put("userBarcode", user != null ? user.getBarcode() : patronIdentifier)
+            .put("servicePointId", sessionData.getScLocation());
+
+        final Map<String, String> headers = getBaseHeaders();
+
+        final CheckoutRequestData checkoutRequestData =
+            new CheckoutRequestData(body, headers, sessionData);
+
+        final Future<IResource> result = resourceProvider.createResource(checkoutRequestData);
+
+        return result
+          .otherwise(Utils::handleErrors)
+          .compose(res -> addTitleIfNotFound(sessionData, itemIdentifier, res))
+          .map(resource -> {
+            final Optional<JsonObject> response = Optional.ofNullable(resource.getResource());
+
+            final OffsetDateTime dueDate = response
+                .map(v -> v.getString("dueDate"))
+                .map(v -> OffsetDateTime.from(Utils.getFolioDateTimeFormatter().parse(v)))
+                .orElse(OffsetDateTime.now(clock));
+
+            return CheckoutResponse.builder()
+              .ok(Boolean.valueOf(response.isPresent()))
+              .renewalOk(FALSE)
+              .magneticMedia(null)
+              .desensitize(Boolean.valueOf(response.isPresent()))
+              .transactionDate(OffsetDateTime.now(clock))
+              .institutionId(institutionId)
+              .patronIdentifier(patronIdentifier)
+              .itemIdentifier(itemIdentifier)
+              .titleIdentifier(response.map(
+                  v -> getChildString(v, "item", TITLE, UNKNOWN))
+                .orElse(resource.getTitle()))
+              .dueDate(dueDate)
+              .screenMessage(Optional.of(resource.getErrorMessages())
+                .filter(v -> !v.isEmpty())
+                .orElse(null))
+              .build();
+
+          });
+      });
+  }
+
+  private Future<IResource> addTitleIfNotFound(SessionData sessionData,
+                                               String itemIdentifier, IResource circRes) {
+    if (circRes.getErrorMessages().isEmpty()) {
+      return Future.succeededFuture(circRes);
+    }
+    return getTitle(itemIdentifier, sessionData, circRes.getErrorMessages());
+  }
+
+  private IResource getiResourceFromTitle(String title, List<String> circErrorMessages) {
+    IResource res;
+    res = new IResource() {
+      @Override
```

Comment:
can do `return new IResource()` instead of assignment.

Reply:
done